From cf3b94cdd6c289a2336a0c18a29b57d2474be20c Mon Sep 17 00:00:00 2001
From: Sergey Temerkhanov <s.temerkhanov@gmail.com>
Date: Thu, 27 Jul 2017 03:37:09 +0300
Subject: [PATCH 0668/1239] cavium: thunderx: Enable PCI-based SCSI AHCI driver

Enable the PCI-based SCSI AHCI driver in the default T83 config

Signed-off-by: Sergey Temerkhanov <s.temerkhanov@gmail.com>
---
 configs/t83_defconfig           |  1 +
 include/configs/thunderx_83xx.h | 12 +++++++-----
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/configs/t83_defconfig b/configs/t83_defconfig
index 3c777a8a9d..c86644b7d1 100644
--- a/configs/t83_defconfig
+++ b/configs/t83_defconfig
@@ -38,6 +38,7 @@ CONFIG_EFI_PARTITION=y
 CONFIG_PARTITION_TYPE_GUID=y
 CONFIG_OF_BOARD=y
 CONFIG_NET_RANDOM_ETHADDR=y
+CONFIG_AHCI_PCI=y
 CONFIG_NVME=y
 CONFIG_MISC=y
 CONFIG_PHYLIB=y
diff --git a/include/configs/thunderx_83xx.h b/include/configs/thunderx_83xx.h
index 3bc7ea50ca..04c4ac1418 100644
--- a/include/configs/thunderx_83xx.h
+++ b/include/configs/thunderx_83xx.h
@@ -7,6 +7,8 @@
 #ifndef __THUNDERX_83XX_H__
 #define __THUNDERX_83XX_H__
 
+#define DEBUG
+
 /** Thunder 83xx does not support NOR flash */
 #define CONFIG_SYS_NO_FLASH
 
@@ -163,11 +165,11 @@
 #define CONFIG_ENV_CALLBACK_LIST_STATIC "txsmi\\d?mode:smimode"
 
 /* AHCI support Definitions */
-#ifdef CONFIG_CMD_SATA
-/** Enable AHCI SATA driver */
-# define CONFIG_SATA_AHCI
-/** Maximum number of SATA devices */
-# define CONFIG_SYS_SATA_MAX_DEVICE	6
+#define CONFIG_SCSI_AHCI
+
+#ifdef CONFIG_SCSI_AHCI
+/** Maximum number of SATA devices per controller*/
+# define CONFIG_SYS_SCSI_MAX_SCSI_ID	1
 /** Enable 48-bit SATA addressing */
 # define CONFIG_LBA48
 /** Enable libata, required for SATA */
-- 
2.29.0

